mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -4157,9 +4159,10 @@ impl OrderbookItem {
         o: OrderbookP2PItem,
         proto_info: BaseRelProtocolInfo,
         conf_info: Option<OrderConfirmationsSettings>,
+        from_pubkey: String,
     ) -> Self {
         OrderbookItem {
-            pubkey: o.pubkey,
+            pubkey: from_pubkey,
```

Comment:
I made the simplest possible change to address the impersonation issues. I don't see any practical difference as we are removing any motivation for a bad actor to send these type of messages.

Reply:
The only problem will be if a user wants to match an order by pubkey and an impersonator changed it
https://github.com/KomodoPlatform/atomicDEX-API/blob/adde21b26cc9fdf2822f92a104c981e7f0c2a8e1/mm2src/mm2_main/src/lp_ordermatch.rs#L1491
This is a very small use case, but we have to be careful if we built anything on top of it in the future. I can accept this as a temporary solution, and will work on another PR next week that ignores keep alive message if the pubkey is different.

Reply:
Do you mean if an impersonator replaced a valid order with a fabricated one? I would consider this a different issue all together, and I think we could come up with a more robust solution to this "uuid is not truly unique" issue that I have mentioned several times.

Reply:
> I think we could come up with a more robust solution to this "uuid is not truly unique" issue that I have mentioned several times.

Agree. I will work on this too next week in the same PR that ignores keep alive message if the pubkey is different, I already have an idea on how to tackle this, it will involve additional checks only at this stage until we have a better solution.